ibc-clients/ics07-tendermint/types/src/header.rs

```diff
@@ -135,10 +137,10 @@ impl Header {
             });
         }
 
-        if self.validator_set.hash() != self.signed_header.header.validators_hash {
+        if self.validator_set.hash_with::<H>() != self.signed_header.header.validators_hash {
             return Err(Error::MismatchValidatorsHashes {
                 signed_header_validators_hash: self.signed_header.header.validators_hash,
-                validators_hash: self.validator_set.hash(),
+                validators_hash: self.validator_set.hash_with::<H>(),
```

Comment:
While at it, perhaps:

```suggestion
        let validator_hash = self.validator_set.hash_with::<H>();
        if validator_hash != self.signed_header.header.validators_hash {
            return Err(Error::MismatchValidatorsHashes {
                signed_header_validators_hash: self.signed_header.header.validators_hash,
                validators_hash,
```